### PR TITLE
rafthttp: report unreachable when dropping messages

### PR DESCRIPTION
**Pull request by yichengq**

### CODE REVIEW

File: rafthttp/stream.go [hunk 1]
```diff
@@ -116,6 +116,7 @@ func (cw *streamWriter) run() {
 				if m.Term > msgAppTerm {
 					cw.resetCloser()
 					heartbeatc, msgc = nil, nil
+					cw.r.ReportUnreachable(m.To)
```
Comment (xiang90) [inline rafthttp/stream.go]: i think we should not report unreachable here. we should retry.

**Comment by xiang90** [comment]
LGTM

File: rafthttp/stream.go (hunk 1 above)
Comment (yichengq) [inline rafthttp/stream.go]: i will add a TODO for it. currently we close the connection and fall to unreachable status.

Comment (yichengq) [inline rafthttp/stream.go]: and report to raft at peer level

**Pull request merged**